deployment/frontend/src/server/api/routers/dataset.ts

```diff
@@ -442,6 +453,23 @@ export const DatasetRouter = createTRPCRouter({
                         )
                     })
                 )
+                for (const existingCollaborator of existingCollaborators) {
```

Comment:
You might want to check this if it will show errors, because I THINK in this scenario it might just send the promises and return immediately, since you dont wait for them to finish it might happen that some call to `deleteCollaborator` throws an Error but the function has already been completed, but im not 100% sure

Reply:
@luccasmmg I just pushed a change to this. I did this instead:
```
                try {
                    await Promise.all(
                        existingCollaborators
                            .filter(
                                (existingCollaborator) =>
                                    !newCollaborators.some(
                                        (newCollaborator) =>
                                            newCollaborator.name ===
                                            existingCollaborator.name
                                    )
                            )
                            .map((existingCollaborator) =>
                                deleteCollaborator(
                                    {
                                        package_id: input.id ?? '',
                                        user_id: existingCollaborator.name,
                                    },
                                    ctx.session
                                )
                            )
                    )
                } catch (e) {
                    console.log(e)
                }
```